src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,355 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+from collections.abc import Generator
+from contextlib import contextmanager
+
+from gt4py import eve
+from gt4py.cartesian import utils
+from gt4py.cartesian.gtc.common import (
+    AxisBound,
+    BuiltInLiteral,
+    CartesianOffset,
+    DataType,
+    FieldAccess,
+    HorizontalInterval,
+    HorizontalMask,
+    LevelMarker,
+    LoopOrder,
+    While,
+)
+from gt4py.cartesian.gtc.definitions import Extent
+from gt4py.cartesian.gtc.oir import (
+    AssignStmt,
+    BinaryOp,
+    CacheDesc,
+    Cast,
+    Decl,
+    FieldDecl,
+    HorizontalExecution,
+    HorizontalRestriction,
+    IJCache,
+    Interval,
+    KCache,
+    Literal,
+    LocalScalar,
+    MaskStmt,
+    NativeFuncCall,
+    ScalarAccess,
+    ScalarDecl,
+    Stencil,
+    Temporary,
+    TernaryOp,
+    UnaryOp,
+    UnboundedInterval,
+    VariableKOffset,
+    VerticalLoop,
+    VerticalLoopSection,
+)
+from gt4py.cartesian.gtc.passes.oir_optimizations.utils import StencilExtentComputer
+from gt4py.eve import codegen
+from gt4py.eve.concepts import SymbolRef
+
+
+class DebugCodeGen(codegen.TemplatedGenerator, eve.VisitorWithSymbolTableTrait):
```

Comment:
- It seems you are not using the SymbolTableTrait, but instead of managing symbols manually. What's the reason?
- You are also not using the TemplatedGenerator, only a normal NodeVisitor as far as I can see.

I could give an intro to both in case there are questions.